services/products.js

```diff
@@ -226,6 +226,14 @@ const duplicateProduct = async (idProduct, newCode) => {
         status     : 'liv'
     };
     doc.code    = newCode;
+    for (const lang of Object.entries(doc.translation)) {
```

Comment:
instead of :
```js
for (const lang of Object.entries(doc.translation)) {
  if (doc.translation[lang[0]].canonical) {
    delete doc.translation[lang[0]].canonical;
  }
}
```
you could write it like that :
```js
// descructuring resulting array from Object.entries
for (const [lang] of Object.entries(doc.translation)) {
  if (doc.translation[lang].canonical) {
    delete doc.translation[lang].canonical;
  }
}```